grails-app/controllers/org/pih/warehouse/order/PurchaseOrderController.groovy

```diff
@@ -25,18 +26,23 @@ class PurchaseOrderController {
 
     def create = {
         Order order = new Order()
```

Comment:
Let's move the logic to a method on the order service so it's easier to unit test.
```
OrderService.createNewPurchaseOrder(Location currentLocation)
```